File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {

        int n = nums.size();
        
        vector <int> prefixSum(n);
        vector <int> suffixSum(n);
        
        prefixSum[0] = nums[0];
        suffixSum[n-1] = nums[n-1];
        
        
        for(int i=1; i<n; i++){
            prefixSum[i] = nums[i] + prefixSum[i-1];
        }
        
        for(int i=n-2; i>=0; i--){
            suffixSum[i] = suffixSum[i+1] + nums[i];
        }
        
        for(int i=0; i<n; i++){
            
            if(prefixSum[i] == suffixSum[i]){
                return i;
            }
            
        }
        return -1;    
    }
};

File: 0724-find-pivot-index/0724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {

        int size = nums.length;
        int[] prefixSum = new int[size];
        int[] suffixSum = new int[size];

        prefixSum[0] = nums[0];
        suffixSum[size-1] = nums[size-1];

        for(int i=1; i<size; i++){
            prefixSum[i] = nums[i] + prefixSum[i-1];
        }

        for(int i=size-2; i>=0; i--){
            suffixSum[i] = nums[i] + suffixSum[i+1];
        }
        int index = size-1;
        for(int i=0; i<size; i++){
            if(prefixSum[i] == suffixSum[i]){
                return i;
            }
        }
        return -1;        
    }
}